## EDA On Pokemon Images
This dataset contains 124x124 images of each Pokemon from generations one through seven. Also included is a CSV with each pokemon's type and subtype. I noticed that some images are of pngs and others are jpgs. I'd like to find the difference if any in the representation of both of these image types when they are converted to numpy arrays. If there is a difference I will write a script to convert the data into a common format, whichever is better suited to the application of deep learning. I also want to make sure that all of the images are the same size.

In [7]:
from matplotlib import image 
from numpy import asarray
import os

In [2]:
dataset_path = "/home/gerardo/Desktop/Projects/Datasets/Pokemon/images"

In [3]:
print(len(os.listdir(dataset_path)))

809


There are 809 images. I wonder if this will be enough to train a decent model.

In [17]:
test = dataset_path + "/" + "araquanid.jpg"
test_image = image.imread(test)
test_image.shape

(120, 120, 3)

In [18]:
test2 = dataset_path + "/" + "absol.png"
test2_image = image.imread(test2)
test2_image.shape

(120, 120, 4)

It appears that a jpg image has 3 color dimensions while a png has 4. I want to make sure that all images are at least the same height and width. I check this below.

In [19]:
#Iterate through all images and check their size, print the filename of those that don't match the standard
for file in os.listdir(dataset_path):
    picture_path = dataset_path + "/" + file
    picture = image.imread(picture_path)
    if (picture.shape) != (120,120,4) and (picture.shape != (120,120,3)):
        print(file)

The above cell returned nothing so all of the images are the same height and width. In an effort to reduce the size of the dimension of the training data and thus the required size of the network I am going to convert all png images into jpeg images for training. This was accomplished with the ```transform.sh``` script. One concern is that the images that were converted to jpeg from png all have a black background while the rest of the images have a white background. It appears that the white background images are a small minority and thus removing them from the dataset might not be too detrimental. For now I will leave them in and see how it affects training.